pkg/k8s/service_cache.go

```diff
@@ -114,11 +114,12 @@ func (s *ServiceCache) GetServiceIP(svcID ServiceID) *loadbalancer.L3n4Addr {
 	s.mutex.RLock()
 	defer s.mutex.RUnlock()
 	svc := s.services[svcID]
-	if svc == nil {
+	if svc == nil || len(svc.FrontendIPs) == 0 {
 		return nil
 	}
+
 	for _, port := range svc.Ports {
-		return loadbalancer.NewL3n4Addr(port.Protocol, svc.FrontendIP, port.Port,
+		return loadbalancer.NewL3n4Addr(port.Protocol, svc.FrontendIPs[0], port.Port,
```

Comment:
Why are we picking the first element of `FrontendIPs`?

Reply:
From the comment of the function, this function is used to return a random L3n4Addr that is externally scoped(all frontend IPs are externally scoped). So we pick the first element as it's mostly consistent with the `service.Spec.ClusterIP`.
https://github.com/cilium/cilium/pull/14914/files/35224e9f8741c6d395b7cf754d6454bc6cddff22#diff-054519b4e1e87efe881596a837f8c8d0bf0c52e47bd19270aace9c9f1c86fc18R110-R112

Reply:
Oh... interesting... In that case perhaps it would be better to only return a for which the IP family is supported by Cilium. For example, if Cilium is running with `ipv4-enable: false` then we need to make sure `svc.FrontendIPs[0]` is *not* a v4 address.

Reply:
Yeah, that totally makes sense. Will update the PR.